src/lib/types.h

```diff
@@ -106,7 +108,18 @@ template <> struct hash<pgp_fingerprint_t> {
         return res;
     }
 };
-} // namespace std
+
+template <> struct hash<pgp_sig_id_t> {
+    std::size_t
+    operator()(pgp_sig_id_t const &sigid) const noexcept
+    {
+        /* since signature id value is hash itself, we may use it's low bytes */
+        size_t res = 0;
+        std::memcpy(&res, sigid.data(), sizeof(res));
```

Comment:
Can we add static_assert that pgp_sig_id_t is larger or equal to `size_t`'s size?

Reply:
Added (as well as to the pgp_fingerprint_t part)